components/PasswordInput.js

```diff
@@ -100,23 +104,9 @@ class PasswordInput extends ValidatedInput {
       (this.props.className || ''),
     ]
 
-    const inputProps = { ...this.props }
-    delete inputProps.onValidate
-    delete inputProps.className
-    delete inputProps.showStrength
-    delete inputProps.showWarnings
-    delete inputProps.showSuggestions
-
-    if (!isValid) {
-      inputProps.pattern = '$.^'
-    }
-
     return (
       <div className={classNames.join(' ')}>
-        <input
-          {...inputProps}
-          ref={_el => this._el = _el}
-          type={showPassword ? 'text' : 'password'} />
+        <input {...this.renderProps} />
```

Comment:
Shouldn't we still be getting the element ref here?

We still need to be able to focus the input when clicking the show/hide password button.

Reply:
Yeah, it’s still being done, it’s just in the renderProps of the ValidatedInput component. 😁

Reply:
Ah cool. I see what you did there!